### PR TITLE
[FIX] mail: fix im_status crash on unknown partner

### PR DESCRIPTION
**Pull request by tsm-odoo**
Before this commit, the im_status notification coming from the bus would lead to a crash if the partner is not yet known. This is due to the fact that the notification handler assumes the partner exists thus writing on undefined.

This PR fixes this issue by inserting the partner instead of manually udpating its im_status. Therefore, the partner is either created or updated with the values received from the server.

### CODE REVIEW

File: addons/mail/static/src/new/core/partner_model.js [hunk 1]
```diff
@@ -4,6 +4,7 @@
  * @typedef Data
  * @property {number} id
  * @property {string} name
+ * @property {string} im_status
```
Comment (phenix-factory) [inline addons/mail/static/src/new/core/partner_model.js]: im status is a fixed set of strings so you could document them like this: `@property {'online'|'away'|'offline'} im_status`

File: addons/mail/static/src/new/core/messaging.js [hunk 2]
```diff
@@ -58,6 +58,7 @@ export class Messaging {
             },
             /** @type {Object.<number, import("@mail/new/core/follower_model").Follower>} **/
             followers: {},
+            /** @type {Partner[]} */
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/messaging.js]: key is an object not an array

File: addons/mail/static/src/new/core/partner_model.js [hunk 3]
```diff
@@ -7,20 +7,35 @@
...
-            return state.partners[data.id];
+            partner = state.partners[data.id];
+        } else {
+            partner = new Partner();
         }
-        let partner = new Partner(data);
         partner._state = state;
         state.partners[data.id] = partner;
         // return reactive version
         partner = state.partners[data.id];
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/partner_model.js]: state assignation should only be done in the "create" case

**Review by seb-odoo — COMMENTED** [review]
Few comments, also it would be nice to add a test to cover it.

**Comment by tsm-odoo** [comment]
@seb-odoo, suggestions applied, test added. Should be ready to merge now :+1:

**Review by seb-odoo — APPROVED** [review]

**Pull request merged**